src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/TableMappingFactory.java

```diff
@@ -44,55 +40,67 @@
-     * @param amazonDynamoDb            the underlying {@code AmazonDynamoDB} delegate
+     * @param partitioningStrategy      how data are partitioned in tables
      * @param createTablesEagerly       a flag indicating whether to create physical tables eagerly at start time
-     * @param pollIntervalSeconds       the interval in seconds between attempts at checking the status of the table
-     *                                  being created
      */
     public TableMappingFactory(CreateTableRequestFactory createTableRequestFactory,
+                               PhysicalTableManager physicalTableManager,
                                MtAmazonDynamoDbContextProvider mtContext,
-                               AmazonDynamoDB amazonDynamoDb,
                                TablePartitioningStrategy partitioningStrategy,
-                               boolean createTablesEagerly,
-                               int pollIntervalSeconds) {
+                               boolean createTablesEagerly) {
         this.createTableRequestFactory = createTableRequestFactory;
+        this.physicalTableManager = physicalTableManager;
         this.mtContext = mtContext;
-        this.amazonDynamoDb = amazonDynamoDb;
-        this.dynamoDbAdminUtils = new AmazonDynamoDbAdminUtils(amazonDynamoDb);
         this.partitioningStrategy = partitioningStrategy;
         this.virtualTableCreationValidator = new VirtualTableCreationValidator(partitioningStrategy);
-        this.pollIntervalSeconds = pollIntervalSeconds;
-        this.physicalTableDescriptions = new ConcurrentHashMap<>();
         if (createTablesEagerly) {
-            createTablesEagerly(createTableRequestFactory);
+            createTablesEagerly(createTableRequestFactory, physicalTableManager);
         }
     }
 
     CreateTableRequestFactory getCreateTableRequestFactory() {
         return createTableRequestFactory;
     }
 
-    private void createTablesEagerly(CreateTableRequestFactory createTableRequestFactory) {
-        createTableRequestFactory.getPhysicalTables().forEach(this::createTableIfNotExists);
+    private static void createTablesEagerly(CreateTableRequestFactory createTableRequestFactory,
+                                            PhysicalTableManager physicalTableManager) {
+        createTableRequestFactory.getStaticPhysicalTables().forEach(physicalTableManager::createTableIfNotExists);
     }
 
-    void validateCreateVirtualTableRequest(CreateTableRequest createVirtualTableRequest) {
-        DynamoTableDescription virtualTable = new DynamoTableDescriptionImpl(createVirtualTableRequest);
-        DynamoTableDescription physicalTable = lookupPhysicalTable(virtualTable);
+    /**
+     * Creates the table mapping for a new virtual table, creating the physical table if it doesn't exist.
+     *
+     */
+    TableMapping createNewTableMapping(CreateTableRequest virtualCreateTableRequest, boolean isMultitenant) {
```

Comment:
Now in addition to the existing getTableMapping(), we also have createNewTableMapping(). createNewTableMapping() is called when MtAmazonDynamoDbBySharedTable.createTable() or createMultitenantTable() is called, and creates the corresponding physical table if needed. getTableMapping() used to create the physical table if it doesn't exist, but now it doesn't -- it only calls describe.

Reply:
Upon createTable, we used to only validate the virtual and physical table definitions, without creating the physical table if it doesn't exist. Now we make sure the physical table is created right away, instead of possibly waiting until the first DML.